src/MSBuild/app.amd64.config

```diff
@@ -172,15 +172,15 @@
         <property name="MSBuildToolsPath32" value="$([MSBuild]::GetToolsDirectory32())" />
         <property name="MSBuildToolsPath64" value="$([MSBuild]::GetToolsDirectory64())" />
         <property name="MSBuildSDKsPath" value="$([MSBuild]::GetMSBuildSDKsPath())" />
-        <property name="FrameworkSDKRoot" value="$([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8', 'InstallationFolder', null, RegistryView.Registry32))" />
+        <property name="FrameworkSDKRoot" value="$([MSBuild]::GetRegistryValueFromView('HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8.1', 'InstallationFolder', null, RegistryView.Registry32))" />
```

Comment:
Is this safe to do? What does the default VS installer install for  the .NET SDK? Is it 4.8.1? If it's not, what happens if 4.8.1 isn't installed?

Reply:
This is what I've been most curious about. https://github.com/dotnet/msbuild/pull/4487 merged with no discussion so I don't have much context on how safe this is. At face value it looks like FrameworkSDKRoot would be empty. I'll try setting up a clean machine with a clean install and see what happens.

What we could safely do is create a "backup" property that stays on 4.8, and if the new prop is empty we fall back to the backup.

Reply:
Hmm. This is the default installer (live public 17.4):

![image](https://user-images.githubusercontent.com/3347530/201734696-38c00bfd-9b73-4322-bb99-264005c230b0.png)

So the user is _not_ guaranteed to have 4.8 installed, and should be guaranteed to have 4.7.2. But I don't remember hearing any grumbling about pointing to 4.8 in the toolset. @tmeschter do you recall any additional context on #4487?

Reply:
When removing the registry entry (in a VM 😱) I verified that calling `GetPathToDotNetFrameworkSdkFile` passing net481, it does return null.

Building a localized resource also shows this error:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Task could not find "al.exe" using the SdkToolsPath "" or the registry key "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\NETFXSDK\4.8\WinSDK-NetFx40Tools-x86". Make sure the SdkToolsPath is set and the tool exists in the correct processor specific location under the SdkToolsPath and that the Microsoft Windows SDK is installed	ToolLocationHelperRepro			
```

So if the key isn't there there _will_ be a build failure, which was entirely possible when we updated to the 4.8 registry key. Maybe we wait until the next major version for that update as well?

Reply:
^ this is also an argument for having the "backup" property where we query for 4.8.1 and 4.8, and if 4.8.1 is empty then we use 4.8